internal/executor/instance/instance.go

```diff
@@ -126,12 +133,42 @@ func (inst *Instance) Run(ctx context.Context, config *RunConfig) error {
 			Memory:   int64(inst.memory * mebi),
 		},
 	}
+
+	// In case the additional containers are used, tell the agent to wait for them
+	if len(inst.additionalContainers) > 0 {
+		var ports []string
+		for _, additionalContainer := range inst.additionalContainers {
+			ports = append(ports, strconv.FormatUint(uint64(additionalContainer.ContainerPort), 10))
+		}
+		commaDelimitedPorts := strings.Join(ports, ",")
+		containerConfig.Env = append(containerConfig.Env, "CIRRUS_PORTS_WAIT_FOR="+commaDelimitedPorts)
+	}
+
 	cont, err := cli.ContainerCreate(ctx, &containerConfig, &hostConfig, nil, "")
 	if err != nil {
 		return err
 	}
 
+	// Start additional containers (if any)
+	var additionalContainersWG sync.WaitGroup
+	additionalContainersErrChan := make(chan error, len(inst.additionalContainers))
+	additionalContainersCtx, additionalContainersCancel := context.WithCancel(context.Background())
+	for _, additionalContainer := range inst.additionalContainers {
+		additionalContainer := additionalContainer
+
+		additionalContainersWG.Add(1)
+		go func() {
+			if err := runAdditionalContainer(additionalContainersCtx, logger, additionalContainer, cli, cont.ID); err != nil {
+				additionalContainersErrChan <- err
+			}
+			additionalContainersWG.Done()
+		}()
+	}
+
 	defer func() {
+		additionalContainersCancel()
```

Comment:
With that logic I think cancalation of the additional containers context should be after removing the main container. 

What do you think about moving this defer right after declaration of `additionalContainersCancel`?

Reply:
> With that logic I think cancalation of the additional containers context should be after removing the main container.

See https://github.com/cirruslabs/cirrus-cli/pull/15/commits/467c8f3159f558764b43a5c90c891c5700912607.

> What do you think about moving this defer right after declaration of `additionalContainersCancel`?

See https://github.com/cirruslabs/cirrus-cli/pull/15/commits/b67ac6007febf1b02e2a5e1fc47758638cb03d0a.